### ISSUE TITLE
[Vanity URL] : Missing parameters in Vanity URL cause problems in API

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#20685]
Reported via Support: [104523](https://dotcms.zendesk.com/agent/tickets/104523)
Reproducible in dotCMS 5.3.8.4LTS, up to latest master (21.06)

Because of an incorrect content import, or problems during the import, Vanity URLs can make it to the database without having required fields, such as:

1. Title
2. Forward To

In customer environment with thousands of Vanity URLs, it's almost impossible to manually find the offending piece of content. Troubleshooting this can get even worse when the error in the log does not indicate the exact Vanity URL that is causing problems.

**To Reproduce**
1. Create test HTML Pages in both English and Spanish.
2. Using the Demo Starter, create a Vanity URL **that only points to the Spanish page** using the `?language_id` parameter.
3. In the English page, add a `Simple Widget` content with an HTML link that points to the Spanish Vanity URL you created before.
4. Flush all the dotCMS cache regions.
5. Now, directly in the database, set the `Uri` value of the Vanity URL to an empty String. For example, the Vanity URL can look like this:

<img width="1417" alt="conflicting-vanity" src="https://user-images.githubusercontent.com/8779585/126203148-8d262fe8-18b2-4690-bef9-8c70f71663ef.png">

5. Open up an Incognito Window with the English page, and click the link to the Spanish page. The following error will be displayed in the `dotcms.log` file:
```
13:24:46.557  ERROR business.VanityUrlAPIImpl - An error occurred when retrieving Vanity URLs: siteId=[SYSTEM_HOST], languageId=[1584454943708]
java.lang.NullPointerException: null
13:24:46.558  ERROR lang.Class - null
com.dotmarketing.business.DotStateException: null
	at com.dotcms.vanityurl.business.VanityUrlAPIImpl.findInDb_aroundBody0(VanityUrlAPIImpl.java:136) ~[dotcms_5.3.8.4_999999.jar:?]
	at com.dotcms.vanityurl.business.VanityUrlAPIImpl$AjcClosure1.run(VanityUrlAPIImpl.java:1) ~[dotcms_5.3.8.4_999999.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.8.4_999999.jar:?]
...
...
```

**Expected behavior**
Any problem coming from the Vanity URL matching process must indicate the offending piece of content.

**Additional context**
The offending piece of content could be found only by adding a Debug Plugin. This was specially complicated because the customer is running on a Docker environment.

**Comment by wezell** [comment]
We need to make the list load and only reject and log the ones that are not valid

**Comment by nollymar** [comment]
Passed Internal QA

**Comment by bryanboza** [comment]
Fixed, tested on release-21.09 // Postgres // FF
